src/CommunityToolkit.Maui.Core/Views/Popup/PopupExtensions.windows.cs

```diff
@@ -88,16 +91,48 @@ public static void SetSize(this Popup mauiPopup, IPopup popup)
 			}
 		}
 
-		if (popup.Parent is IView popupParent)
-		{
-			currentSize.Width = Math.Min(currentSize.Width, popupParent.Frame.Width);
-			currentSize.Height = Math.Min(currentSize.Height, popupParent.Frame.Height);
-		}
-		
+		currentSize.Width = Math.Min(currentSize.Width, popupParent.Bounds.Width);
+		currentSize.Height = Math.Min(currentSize.Height, popupParent.Bounds.Height);
+
 		mauiPopup.Width = currentSize.Width;
 		mauiPopup.Height = currentSize.Height;
 		mauiPopup.MinWidth = mauiPopup.MaxWidth = currentSize.Width + (defaultBorderThickness * 2);
 		mauiPopup.MinHeight = mauiPopup.MaxHeight = currentSize.Height + (defaultBorderThickness * 2);
+
+		if (mauiPopup.Child is FrameworkElement control)
+		{
+			var verticalOptions = popup.VerticalOptions;
+			var horizontalOptions = popup.HorizontalOptions;
+
+			if (IsFillLeft(verticalOptions, horizontalOptions) || IsFillCenter(verticalOptions, horizontalOptions) || IsFillRight(verticalOptions, horizontalOptions))
+			{
+				control.Width = mauiPopup.Width;
+				control.Height = popupParent.Bounds.Height;
+			}
+			else if (IsTopFill(verticalOptions, horizontalOptions) || IsCenterFill(verticalOptions, horizontalOptions) || IsBottomFill(verticalOptions, horizontalOptions))
+			{
+				control.Width = popupParent.Bounds.Width;
+				control.Height = mauiPopup.Height;
+			}
+			else if (IsFill(verticalOptions, horizontalOptions))
+			{
+				control.Width = popupParent.Bounds.Width;
+				control.Height = popupParent.Bounds.Height;
+			}
+			else
+			{
+				control.Width = mauiPopup.Width;
+				control.Height = mauiPopup.Height;
+			}
+
+			static bool IsFillLeft(LayoutAlignment verticalOptions, LayoutAlignment horizontalOptions) => verticalOptions == LayoutAlignment.Fill && horizontalOptions == LayoutAlignment.Start;
```

Comment:
the code seems to be the same as on line 270

Reply:
The IsFillLeft, IsFillCenter, IsFillRight, IsTopFill, IsCeneterFll, IsBottomFill, and IsFill methods were defined within the SetSize and SetLayout methods, respectively, so I fixed them by redefining them as class methods.

Reply:
Since the operation is different from Android and iOS, I will reconsider the operation when Fill is specified for VerticalOptions and HorizontalOptions of Popup. At first glance, I thought it was good, but only Windows behaved differently. This is because even if an explicit size is given to the Popup, it will be resized regardless of that size.

Reply:
https://github.com/CommunityToolkit/Maui/pull/1361#issuecomment-1688296364
I would like to verify the operation of this point on Windows as well.
I will make corrections as necessary, so please wait for a while.

Reply:
I made a change so that LayoutOptions are respected when the Popup size is unspecified and the 
Popup's Content size is unspecified. Based on the post below.
https://github.com/CommunityToolkit/Maui/pull/1361#issuecomment-1688296364

Below are the changes.

[src\CommunityToolkit.Maui.Core\Views\Popup\PopupExtensions.windows.cs]

    public static void SetSize(this Popup mauiPopup, IPopup popup, IMauiContext? mauiContext)
    {
        ArgumentNullException.ThrowIfNull(mauiContext);
        ArgumentNullException.ThrowIfNull(popup.Content);

        const double defaultBorderThickness = 0;
        const double defaultSize = 600;

        var popupParent = mauiContext.GetPlatformWindow();
        var standardSize = new Size { Width = defaultSize, Height = defaultSize / 2 };

        Size currentSize = standardSize;

        if (!popup.Size.IsZero)
        {
            currentSize.Width = popup.Size.Width;
            currentSize.Height = popup.Size.Height;
        }
        else
        {
            if (double.IsNaN(popup.Content.Width) || (double.IsNaN(popup.Content.Height)))
            {
                currentSize = popup.Content.Measure(popupParent.Bounds.Width, popupParent.Bounds.Height);

                if (double.IsNaN(popup.Content.Width))
                {
                    currentSize.Width = popup.HorizontalOptions == LayoutAlignment.Fill ? popupParent.Bounds.Width : currentSize.Width;
                }
                if (double.IsNaN(popup.Content.Height))
                {
                    currentSize.Height = popup.VerticalOptions == LayoutAlignment.Fill ? popupParent.Bounds.Height : currentSize.Height;
                }
            }
            else
            {
                currentSize.Width = popup.Content.Width;
                currentSize.Height = popup.Content.Height;
            }
        }

        currentSize.Width = Math.Min(currentSize.Width, popupParent.Bounds.Width);
        currentSize.Height = Math.Min(currentSize.Height, popupParent.Bounds.Height);

        mauiPopup.Width = currentSize.Width;
        mauiPopup.Height = currentSize.Height;
        mauiPopup.MinWidth = mauiPopup.MaxWidth = currentSize.Width + (defaultBorderThickness * 2);
        mauiPopup.MinHeight = mauiPopup.MaxHeight = currentSize.Height + (defaultBorderThickness * 2);

        if (mauiPopup.Child is FrameworkElement control)
        {
            control.Width = mauiPopup.Width;
            control.Height = mauiPopup.Height;
        }
    }

Verified whether LayoutOptions are respected when Popup size is not specified.
The layout of the Popup used for verification is as follows.

    <toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
        xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
        xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
        x:Class="MauiComm_IssuePopupWindows.PopupPage1">

        <Label Text="1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ" TextColor="Black" HorizontalOptions="Start" />

    </toolkit:Popup>

Below are the verification results on Windows.

[When VerticalOptions is Start]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/23885dc4-e6e7-4ad4-b5e3-eb1a929c3d71" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/ca597ae5-b0c1-4bb9-9b53-2848cbab5869" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/d62250a6-ba5f-451c-91c0-db49deb9bebe" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/ef753d81-57d1-4913-96b7-e3da19a01f05" /></td>
  </tr>
</table>

[When VerticalOptions is Center]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/de960f61-20f8-45d2-b863-466044fcad2e" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/8f3c3084-56a9-4410-a3d6-31ea98025fac" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/2210c391-086a-443e-9a5b-11b0de9185dd" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/415f95e9-c1c1-45b1-b067-eb54ef499e23" /></td>
  </tr>
</table>

[When VerticalOptions is End]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/1d4e3170-b54c-49ce-acb0-8c398e1f1d0b" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/1521eb4d-e035-4a5f-9e1c-d8b4d32bd9e8" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/2fe5c705-1d85-47a1-a1d5-23dacbb46f7e" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/9fcfe3f3-8570-477e-8c3a-bb7a89ad806a" /></td>
  </tr>
</table>

[When VerticalOptions is Fill]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/1ba30a19-48c5-4104-87db-42aa29b14f4c" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/522dd547-3043-4a97-8e1d-176c41d6b972" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/f91123fe-d7d3-4e80-af9f-e37d380aab24" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/f0c0e8ad-8a51-45e7-bf82-7265c825bef1" /></td>
  </tr>
</table>

You can see that LayoutOptions are respected when the Popup size is unspecified and the Popup 
Content size is unspecified.

Below are additional validation results.

    <toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
        xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
        xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
        x:Class="MauiComm_IssuePopupWindows.PopupPage2">

        <Grid WidthRequest="100">
            <Grid WidthRequest="10" HeightRequest="10" VerticalOptions="Start" HorizontalOptions="Start" BackgroundColor="Blue" />
            <Grid WidthRequest="10" HeightRequest="10" VerticalOptions="Start" HorizontalOptions="End" BackgroundColor="Blue" />
            <Grid WidthRequest="10" HeightRequest="10" VerticalOptions="End" HorizontalOptions="Start" BackgroundColor="Blue" />
            <Grid WidthRequest="10" HeightRequest="10" VerticalOptions="End" HorizontalOptions="End" BackgroundColor="Blue" />
        </Grid>

    </toolkit:Popup>

Below are the verification results on Windows.

[When VerticalOptions is Start]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/2f837ff8-6950-4e8a-9398-d76b1585cd7f" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/cf61a33b-9e58-47b0-a46b-53c8ae24adc3" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/6dc92f70-871c-4add-a6af-1162d812a23e" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/3f5e6e05-96df-4671-b8a7-e00b0b572da7" /></td>
  </tr>
</table>

[When VerticalOptions is Center]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/1e2f5e33-a6bf-47c6-a021-992c6de1cbd5" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/f12a2e3e-0408-4f3b-8d62-3e99f084d895" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/623b102f-094d-44ce-9cc2-792bdfb9ff27" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/1ed54d09-cf99-453a-a248-516a384eb67e" /></td>
  </tr>
</table>

[When VerticalOptions is End]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/69f75a60-8034-4e24-8440-e0d2ad61d0bf" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/e96a5f0b-b3e3-4880-9b4d-5cd05aa32074" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/b1829d72-cb25-48ae-b75b-05d445dd003c" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/331f618e-a610-4519-a93f-54d507e4a1cd" /></td>
  </tr>
</table>

[When VerticalOptions is Fill]
I changed the HorizontalOptions to:

<table>
  <tr>
    <td>[Start]</td>
    <td>[Center]</td>
    <td>[End]</td>
    <td>[Fill]</td>
  </tr>
  <tr>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/c1563150-f407-4185-9f35-14dd61d5a212" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/0cbdf4dc-f973-4930-bbee-831aaa6c8316" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/8028723b-5183-4281-b2a5-aae28308ae9d" /></td>
    <td><img width="200" src="https://github.com/CommunityToolkit/Maui/assets/125236133/c26102ba-b943-4f70-a38b-eb08f5b80386" /></td>
  </tr>
</table>

You can see that the VerticalOptions are respected because the height is unspecified.
I won't post the test results, but it will now respect HorizontalOptions when width is unspecified.

And, I laid out the Popup like this:

    <toolkit:Popup xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
        xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
        xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
        x:Class="MauiComm_VerifyPopupStyle.Popup6" VerticalOptions="Center" HorizontalOptions="Start">

        <VerticalStackLayout>
            <Label Text="Popup" FontSize="20"/>
            <BoxView HeightRequest="3" Color="AliceBlue" />        
            <Label x:Name="txt1" Text="Short String" />
            <Label x:Name="txt2" Text="Long String1 Long String2 Long String3 Long String4 Long String5 Long String6 Long String7 Long String8 Long String9 Long String10 Long String" />
        </VerticalStackLayout>

    </toolkit:Popup>

Below is the execution result on Windows.

<img width="250" src="https://github.com/CommunityToolkit/Maui/assets/125236133/2895a781-8449-4230-a329-b44720290064" />

You can see that the label is wrapped like the Android and iOS verification results of PR #1361.